src/ui/components/Preset.tsx

```diff
@@ -21,13 +22,40 @@ const styles = stylesheet.createThemedStyleSheet({
 	}
 });
 
+const { fetchProfile } = findByProps('fetchProfile');
+
 const UserStore = findByStoreName('UserStore');
 
+const defaultAvatars = [
+	getAssetIDByName('default_avatar_0'),
+	getAssetIDByName('default_avatar_1'),
+	getAssetIDByName('default_avatar_2'),
+	getAssetIDByName('default_avatar_3'),
+	getAssetIDByName('default_avatar_4'),
+	getAssetIDByName('default_avatar_5')
+];
+
 // Make this return default avatars for null users
-function renderAvatar(user) {
-	if (user) return <Avatar user={user} size="size16" />
-	else return 
-};
+function renderAvatar({ user, id }) {
+	if (user) return <Avatar user={user} size="size16" />;
+	else {
+		const defaultAvatarIndex = Number((BigInt(id) >> 22n) % 6n);
+		console.log(id, defaultAvatarIndex);
+		return (
+			<Image
+				source={defaultAvatars[defaultAvatarIndex]}
+				style={{
+					width: 16,
+					height: 16,
+					borderRadius: 8
+				}}
+			/>
+		);
+	}
+}
+
+// TODO do this the Fiery way (use like a store or something idk)
+const hasFetchedBefore = {};
```

Comment:
is this okay

Reply:
this scares me

I would just not bother keeping track of if they've fetched before also why are you using fetchProfile? don't you need fetchUser

Reply:
fetchProfile also fetches the user (fetchUser doesnt exist)

Reply:
https://github.com/Vendicated/Vencord/blob/a9e67e2955dfa07d734361c66ceabb0278ec5e34/src/webpack/common/utils.ts#L96
damn you discord. this module is probably findable on mobile with same props (this getUser actually gets it from the api not a store)